core-bundle/src/Routing/AbstractPageRouteProvider.php

```diff
@@ -44,7 +44,7 @@ public function __construct(ContaoFramework $framework, CandidatesInterface $can
      */
     protected function findCandidatePages(Request $request): array
     {
-        $candidates = $this->candidates->getCandidates($request);
+        $candidates = array_map('strval', $this->candidates->getCandidates($request));
```

Comment:
Why are you casting the values to string here? They will be recasted to int in line 58.

Reply:
Because `preg_match` expects a string. We could also cast them in line 57 but that wouldn't remove the second typecast in line 58.